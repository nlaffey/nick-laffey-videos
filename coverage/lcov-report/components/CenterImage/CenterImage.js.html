<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/CenterImage/CenterImage.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">components/CenterImage</a> CenterImage.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">83.33% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">75% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>3/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">83.33% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/6</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
&nbsp;
class centerImage extends React.Component {
&nbsp;
    componentWillMount() {
        if (typeof this.props.image !== "string") {
            throw Error('The image requires a URL string');
        }
    }
&nbsp;
    getAlt() {
        <span class="missing-if-branch" title="if path not taken" >I</span>if (this.props.name) {
<span class="cstat-no" title="statement not covered" >            return this.props.name + ' Juggling Video'</span>
        } else {
            return "Nick Laffey juggling";
        }
    }
&nbsp;
    render() {
        return &lt;div&gt;
            &lt;img className="center-image" src={this.props.image} alt={this.getAlt()}
                 width="960"
                 height="562"/&gt;
        &lt;/div&gt;
    }
}
&nbsp;
export default centerImage;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="http://istanbul-js.org/" target="_blank">istanbul</a> at Mon Sep 19 2016 22:17:58 GMT-0400 (EDT)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
