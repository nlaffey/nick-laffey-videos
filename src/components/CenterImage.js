import React from 'react';

class centerImage extends React.Component {
    render() {
        return <div className="wrapper">
            <img className="pic" src={"src/img/" + this.props.image} alt={this.props.name + ' Juggling Video'}
                 width="960"
                 height="562"/>
        </div>
    }
}

export default centerImage;