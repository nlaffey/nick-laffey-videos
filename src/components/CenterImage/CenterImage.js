import React from 'react';

class centerImage extends React.Component {

    componentWillMount() {
        if (typeof this.props.image !== "string") {
            throw Error('The image requires a URL string');
        }
    }

    getAlt() {
        if (this.props.name) {
            return this.props.name + ' Juggling Video'
        } else {
            return "Nick Laffey juggling";
        }
    }

    render() {
        return <div>
            <img className="center-image" src={this.props.image} alt={this.getAlt()}
                 width="960"
                 height="562"/>
        </div>
    }
}

export default centerImage;