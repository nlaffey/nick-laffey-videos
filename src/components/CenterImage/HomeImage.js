import React from 'react';
import CenterImage from './CenterImage';
import image from '../../img/mondo.jpg';

class InitialImage extends React.Component{
    render(){
        return <div>
            <CenterImage image={image}/>
                <div className="info"><b/>
                    Select a video on the right
                </div>
            </div>
    }
}

export default InitialImage;