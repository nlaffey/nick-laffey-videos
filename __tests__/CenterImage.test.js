import renderer from 'react-test-renderer';
import React from 'react';
import CenterImage from '../src/components/CenterImage/CenterImage';
import image from '../src/img/letitdie.png';

it('renders correctly with image passed in', ()=> {
    const video = renderer.create(
        <CenterImage image={image}/>
    ).toJSON();
    expect(video).toMatchSnapshot();
});

it('throws an error when a non string URL is passed', ()=> {
    expect(()=> {
        renderer.create(
            <CenterImage image={123}/>)
    }).toThrow(new Error('The image requires a URL string'))
});

it('throws an error when no URL is passed', ()=> {
    expect(()=> {
        renderer.create(
            <CenterImage/>
        )
    }).toThrow(new Error('The image requires a URL string'))
});
