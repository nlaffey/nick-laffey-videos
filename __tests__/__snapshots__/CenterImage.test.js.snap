exports[`test renders correctly with image passed in 1`] = `
<div>
  <img
    alt="Nick Laffey juggling"
    className="center-image"
    height="562"
    src="mock_file_name"
    width="960" />
</div>
`;
